titus-server-master/src/main/java/com/netflix/titus/master/clusteroperations/ClusterAgentAutoScaler.java

```diff
@@ -419,20 +426,19 @@ Completable doAgentScaling() {
 
             List<AgentInstance> removableInstancesInInstanceGroup = instancesForActiveInstanceGroupsById.getOrDefault(instanceGroup.getId(), emptyList())
                     .stream()
-                    .filter(i -> i.getOverrideStatus().getState() == InstanceOverrideState.Removable)
+                    .filter(i -> i.getAttributes().containsKey(AgentAttributes.REMOVABLE))
                     .collect(Collectors.toList());
 
             List<AgentInstance> agentsEligibleToRemoveInInstanceGroup = idleInstancesByInstanceGroup.getOrDefault(instanceGroup.getId(), emptyList());
             int agentCountEligibleToRemoveInInstanceGroup = instanceGroup.getCurrent() - instanceGroup.getMin() - removableInstancesInInstanceGroup.size();
             int agentCountToRemoveInInstanceGroup = Ints.min(remainingAgentsToRemove, agentCountEligibleToRemoveInInstanceGroup, agentsEligibleToRemoveInInstanceGroup.size());
             for (int i = 0; i < agentCountToRemoveInInstanceGroup; i++) {
                 AgentInstance agentInstance = agentsEligibleToRemoveInInstanceGroup.get(i);
-                InstanceOverrideStatus removableOverrideStatus = InstanceOverrideStatus.newBuilder()
-                        .withState(InstanceOverrideState.Removable)
-                        .withTimestamp(clock.wallTime())
-                        .withDetail("ClusterAgentAutoScaler setting to Removable to scale down the instance")
-                        .build();
-                Completable completable = agentManagementService.updateInstanceOverride(agentInstance.getId(), removableOverrideStatus);
+                Map<String, String> updatedAttributes = CollectionsExt.merge(
```

Comment:
This is an example, where doing full attributes override at the GRPC layer is dangerous, and prone to race conditions.

Reply:
I agree. Will fix in a new PR.